### ISSUE TITLE
Serial output in 13.0.4 misses data printing

### ISSUE THREAD

**Issue by friebi** [jomjol/AI-on-the-edge-device#1504]
### The Problem

Thanks a lot for your hard work in your freetime and I'm sorry I have to raise another issue 😆 

As you might know from my previous [issue](https://github.com/jomjol/AI-on-the-edge-device/issues/784), I am reading the water meter data from the serial output.
With 13.0.4 I am getting
```
I (324935) TFLITE: task_autodoFlow - next round - Round #1
E (324935) MQTT IF: Init with no Client_ID or Topic. Abort Init!
E (324955) MQTT IF: Init failed, skipping all MQTT publishings in this round!
I (390845) LOGFILE: Remove old log files
I (390855) LOGFILE: Remove old data files
I (390855) TFLITE: task_autodoFlow - round #1 done
```

I thought you might have lowered the severity of the debug output. So I lowered the log level from WARNING to INFO. But when I do that, I end up in the same behavior as described in #1477 (also same logs) and the device does not work at all.

It is totally clear to me that the format etc of the serial output might change without notice as it is not an official interface, but currently it does not work at all for me.

Thanks for your help!

### Version

13.0.4

### Logfile

```shell
[    7604] 2022-12-06T20:14:28 <ERR> [MQTT IF] Init with no Client_ID or Topic. Abort Init!
[    7604] 2022-12-06T20:14:28 <ERR> [MQTT IF] Init failed, skipping all MQTT publishings in this round!
[    7784] 2022-12-06T20:17:29 <ERR> [MQTT IF] Init with no Client_ID or Topic. Abort Init!
[    7784] 2022-12-06T20:17:29 <ERR> [MQTT IF] Init failed, skipping all MQTT publishings in this round!
[       4] 2022-12-06T19:19:25 <INF> [MAIN] =================================================
[       4] 2022-12-06T19:19:25 <INF> [MAIN] ==================== Startup ====================
[       4] 2022-12-06T19:19:25 <INF> [MAIN] =================================================
[       4] 2022-12-06T19:19:25 <INF> [MAIN] Branch: 'HEAD', Revision: 961662f, Date/Time: 2022-12-04 14:50, Web UI: Release: v13.0.4 (Commit: 961662f)
[       4] 2022-12-06T19:19:25 <INF> [MAIN] Reset reason: Via esp_restart
[       9] 2022-12-06T19:19:30 <INF> [MAIN] =================================================
[       9] 2022-12-06T19:19:30 <INF> [MAIN] ================== Main Started =================
[       9] 2022-12-06T19:19:30 <INF> [MAIN] =================================================
[      13] 2022-12-06T19:19:34 <ERR> [MAIN] Camera Framebuffer cannot be initialized!
[      16] 2022-12-06T19:19:36 <ERR> [MAIN] Initialization failed. Will restart in 5 minutes!
[     147] 2022-12-06T19:21:48 <INF> [OTA] Reboot triggered by Software (5s).
[     147] 2022-12-06T19:21:48 <WRN> [OTA] Reboot in 5sec
[       4] 2022-12-06T19:25:05 <INF> [MAIN] =================================================
[       4] 2022-12-06T19:25:05 <INF> [MAIN] ==================== Startup ====================
[       4] 2022-12-06T19:25:05 <INF> [MAIN] =================================================
[       4] 2022-12-06T19:25:05 <INF> [MAIN] Branch: 'HEAD', Revision: 961662f, Date/Time: 2022-12-04 14:50, Web UI: Release: v13.0.4 (Commit: 961662f)
[       4] 2022-12-06T19:25:05 <INF> [MAIN] Reset reason: Via esp_restart
[       9] 2022-12-06T19:25:11 <INF> [MAIN] =================================================
[       9] 2022-12-06T19:25:11 <INF> [MAIN] ================== Main Started =================
[       9] 2022-12-06T19:25:11 <INF> [MAIN] =================================================
[      12] 2022-12-06T19:25:13 <INF> [MAIN] Initialization completed successfully!
[      16] 2022-12-06T20:25:17 <ERR> [MQTT IF] Init with no Client_ID or Topic. Abort Init!
[      16] 2022-12-06T20:25:17 <ERR> [MQTT IF] Init failed, skipping all MQTT publishings in this round!
[     196] 2022-12-06T20:28:18 <ERR> [MQTT IF] Init with no Client_ID or Topic. Abort Init!
[     196] 2022-12-06T20:28:18 <ERR> [MQTT IF] Init failed, skipping all MQTT publishings in this round!
[       4] 2022-12-06T19:30:58 <INF> [MAIN] =================================================
[       4] 2022-12-06T19:30:58 <INF> [MAIN] ==================== Startup ====================
[       4] 2022-12-06T19:30:58 <INF> [MAIN] =================================================
[       4] 2022-12-06T19:30:58 <INF> [MAIN] Branch: 'HEAD', Revision: 961662f, Date/Time: 2022-12-04 14:50, Web UI: Release: v13.0.4 (Commit: 961662f)
[       4] 2022-12-06T19:30:58 <INF> [MAIN] Reset reason: Exception/panic
[      17] 2022-12-06T19:31:11 <INF> [MAIN] =================================================
[      17] 2022-12-06T19:31:11 <INF> [MAIN] ================== Main Started =================
[      17] 2022-12-06T19:31:11 <INF> [MAIN] =================================================
[      20] 2022-12-06T19:31:13 <INF> [MAIN] Initialization completed successfully!
[      20] 2022-12-06T19:31:13 <WRN> [TFLITE] Restarted due to an Exception/panic! Postponing first round start by 5 minutes to allow for an OTA or to fetch the log!
[      20] 2022-12-06T19:31:13 <WRN> [TFLITE] Setting logfile level to DEBUG until the next reboot!
[     298] 2022-12-06T19:35:51 <DBG> [OTA FILE] download_get_handler
[     299] 2022-12-06T19:35:53 <DBG> [OTA FILE] download_get_handler
[     323] 2022-12-06T20:36:16 <ERR> [MQTT IF] Init with no Client_ID or Topic. Abort Init!
[     323] 2022-12-06T20:36:16 <ERR> [MQTT IF] Init failed, skipping all MQTT publishings in this round!
[     503] 2022-12-06T20:39:17 <ERR> [MQTT IF] Init with no Client_ID or Topic. Abort Init!
[     503] 2022-12-06T20:39:17 <ERR> [MQTT IF] Init failed, skipping all MQTT publishings in this round!
```

### Expected Behavior

Be able to extract meter reading from serial output.

### Additional Context

This is how the log (the part relevant for me) used to look like with 12.0.1 and earlier:
```
Result General(Analog)0 - CCW: 0 -  9.157520
FlowControll.doFlow - ClassFlowPostProcessing
Anzahl NUMBERS: 2
UpdateNachkommaDecShift NUMBER0: Nachkomma 0, DecShift 0
UpdateNachkommaDecShift NUMBER1: Nachkomma -3, DecShift 4
ClassFlowCNNGeneral::getReadout _analog=0, _extendedResolution=0, prev=-1
ClassFlowCNNGeneral::ZeigerEvalHybridNeu - kein Vorg�nger - Ergebnis = 7 zahl: 6.996997 zahl_vorgaenger = -1.000000 eval_vorgaenger = -1 Di
gitalUnschaerfe = 0.200000
ClassFlowCNNGeneral::getReadout(dig100)  prev=7
ClassFlowCNNGeneral::ZeigerEvalHybridNeu - KEIN Analoger Vorg�nger, kein Ziffernwechsel, da Vorkomma weit genug weg = 2 zahl: 2.016850 zahl
_vorgaenger = 6.996997 eval_vorgaenger = 7 DigitalUnschaerfe = 0.200000
ClassFlowCNNGeneral::getReadout#ZeigerEvalHybridNeu()= 2
ClassFlowCNNGeneral::getReadout#result= 27
ClassFlowCNNGeneral::ZeigerEvalHybridNeu - KEIN Analoger Vorg�nger, kein Ziffernwechsel, da Vorkomma weit genug weg = 1 zahl: 1.001287 zahl
_vorgaenger = 2.016850 eval_vorgaenger = 2 DigitalUnschaerfe = 0.200000
ClassFlowCNNGeneral::getReadout#ZeigerEvalHybridNeu()= 1
ClassFlowCNNGeneral::getReadout#result= 127
ClassFlowCNNGeneral::ZeigerEvalHybridNeu - KEIN Analoger Vorg�nger, kein Ziffernwechsel, da Vorkomma weit genug weg = 0 zahl: 0.001393 zahl
_vorgaenger = 1.001287 eval_vorgaenger = 1 DigitalUnschaerfe = 0.200000
ClassFlowCNNGeneral::getReadout#ZeigerEvalHybridNeu()= 0
ClassFlowCNNGeneral::getReadout#result= 0127
ClassFlowCNNGeneral::ZeigerEvalHybridNeu - KEIN Analoger Vorg�nger, Nulldurchgang hat stattgefunden = 0 zahl: 0.004251 zahl_vorgaenger = 0.
001393 eval_vorgaenger = 0 DigitalUnschaerfe = 0.200000
ClassFlowCNNGeneral::getReadout#ZeigerEvalHybridNeu()= 0
ClassFlowCNNGeneral::getReadout#result= 00127
PostProcessing - Raw: 00127 Value: 127 Error: no error
UpdateNachkommaDecShift NUMBER0: Nachkomma 0, DecShift 0
UpdateNachkommaDecShift NUMBER1: Nachkomma -3, DecShift 4
ClassFlowCNNGeneral::getReadout _analog=1, _extendedResolution=0, prev=-1
ClassFlowCNNGeneral::ZeigerEvalHybridNeu - kein Vorg�nger - Ergebnis = 3 zahl: 3.301548 zahl_vorgaenger = 9.157520 eval_vorgaenger = -1 Dig
italUnschaerfe = 0.200000
ClassFlowCNNGeneral::getReadout(dig100)  prev=3
ClassFlowCNNGeneral::ZeigerEvalHybridNeu - KEIN Analoger Vorg�nger, kein Ziffernwechsel, da Vorkomma weit genug weg = 4 zahl: 4.280873 zahl
_vorgaenger = 3.301548 eval_vorgaenger = 3 DigitalUnschaerfe = 0.200000
ClassFlowCNNGeneral::getReadout#ZeigerEvalHybridNeu()= 4
ClassFlowCNNGeneral::getReadout#result= 43
ClassFlowCNNGeneral::ZeigerEvalHybridNeu - KEIN Analoger Vorg�nger, kein Ziffernwechsel, da Vorkomma weit genug weg = 4 zahl: 4.304952 zahl
_vorgaenger = 4.280873 eval_vorgaenger = 4 DigitalUnschaerfe = 0.200000
ClassFlowCNNGeneral::getReadout#ZeigerEvalHybridNeu()= 4
ClassFlowCNNGeneral::getReadout#result= 443
Write PreValue Zeile: main      2022-10-11T18:28:00     127

Write PreValue Zeile: sub       2022-10-11T14:48:23     10000

doflow - end 20221011-182752
Remove older log files
I (2461021) log: 0 older log files deleted. 2 current log files not deleted.
task_autodoFlow - round done
```

**Comment by jomjol** [comment]
Currently I suggest to go back to an old version. As the major improvement is the MQTT connection, you will loose nothing while staying within the old version.

**Comment by friebi** [comment]
Yes, I agree this issue has low Prio, but I wanted to let you know so that you are aware.

I will go back to 12.0.1 for now. It would be great if you could mention this issue with a future release when you might have solved it. Such that I know I can try another update 😄 

Take your time and take care!

**Comment by caco3** [comment]
@friebi, can you try if https://github.com/jomjol/AI-on-the-edge-device/actions?query=branch%3Avarious-corrections is ok for you?

It now prints something like this on every round:
```
[0;32mI (96945) FLOW POSTPROC: main: Raw: 051, Value: 51, Status: no error
```

"main" is the name of your Number.

**Comment by friebi** [comment]
Thanks for picking this up!

I have flashed the binaries from your PR and it seems I can now set any log level without running into a boot loop.
However, I'm not able to see the output you are referring. So I so not see the meter reading.

Could be, because I just tested with a second ESP32 on my desk and this does not get any meaningful input picture, of course.
But I would have expected some NaN results at least in the logs.

Unmounting my productive device is always a bit messy :D and I wanted to test it on my desk first.
